squidpy/pl/_spatial.py

```diff
@@ -0,0 +1,316 @@
+from __future__ import annotations
+
+from types import MappingProxyType
+from typing import Any, Tuple, Mapping, Optional, Sequence
+from pathlib import Path
+from functools import partial
+import itertools
+
+from anndata import AnnData
+from scanpy.plotting._utils import _FontSize, _FontWeight
+
+from matplotlib.axes import Axes
+from matplotlib.colors import Colormap
+from matplotlib.figure import Figure
+
+from squidpy.gr._utils import _assert_spatial_basis
+from squidpy.pl._utils import save_fig, _sanitize_anndata
+from squidpy.pl._spatial_utils import (
+    _subs,
+    _SeqStr,
+    _SeqArray,
+    _SeqFloat,
+    Palette_t,
+    _Normalize,
+    _CoordTuple,
+    _plot_edges,
+    _AvailShapes,
+    _set_outline,
+    _decorate_axs,
+    _plot_scatter,
+    _plot_segment,
+    _set_ax_title,
+    _map_color_seg,
+    _set_coords_crops,
+    _prepare_args_plot,
+    _image_spatial_attrs,
+    _prepare_params_plot,
+    _set_color_source_vec,
+)
+from squidpy._constants._constants import ScatterShape
+from squidpy._constants._pkg_constants import Key
+
+
+def spatial_point(
+    adata: AnnData,
+    **kwargs: Any,
+) -> Any:
+    """Spatial shape."""
+    _spatial_plot(adata, img_res_key=None, size_key=None, seg_key=None, **kwargs)
+
+
+def spatial_shape(
+    adata: AnnData,
+    shape: _AvailShapes | None = ScatterShape.CIRCLE.s,  # type: ignore[assignment]
+    img: _SeqArray | bool = True,
+    **kwargs: Any,
+) -> Any:
+    """Spatial shape."""
+    _spatial_plot(adata, shape=shape, img=img, seg_key=None, **kwargs)
+
+
+def spatial_segment(
+    adata: AnnData,
+    cell_id_key: str,
+    seg_key: str = Key.uns.image_seg_key,
+    img: _SeqArray | bool = True,
+    seg: _SeqArray | bool = True,
+    **kwargs: Any,
+) -> Any:
+    """Spatial shape."""
+    _spatial_plot(adata, img=img, seg=seg, seg_key=seg_key, cell_id_key=cell_id_key, **kwargs)
+
+
+def _spatial_plot(
+    adata: AnnData,
+    spatial_key: str = Key.obsm.spatial,
+    shape: _AvailShapes | None = None,
+    color: Sequence[str | None] | str | None = None,
+    groups: _SeqStr = None,
+    library_id: _SeqStr = None,
+    library_key: str | None = None,
+    # features
+    use_raw: bool | None = None,
+    layer: str | None = None,
+    alt_var: str | None = None,
+    # size, coords, cmap, palette
+    size: _SeqFloat = None,
+    size_key: str | None = Key.uns.size_key,
+    scale_factor: _SeqFloat = None,
+    crop_coord: Sequence[_CoordTuple] | _CoordTuple | None = None,
+    cmap: Colormap | str | None = None,
+    palette: Palette_t = None,
+    alpha: Optional[float] = None,
+    norm: _Normalize = None,
+    na_color: str | Tuple[float, ...] = (0, 0, 0, 0),
+    # image
+    img: _SeqArray | bool = None,
```

Comment:
so here and also in `seg` I've implemented type of argument like this in order to use None. Let me explain, I basically want this argument to store both logic for using it as well as its values.

So if the arg is True or array then it's used, if it's False it's not used. It's similar to what we tried to do here:  https://github.com/theislab/scanpy/blob/8736fc3b02a6af3b7214481c2cf78a3f8f5e06bd/scanpy/plotting/_tools/scatterplots.py#L1371

an alternative way that I can come up with is to add an additional argument e.g. `plot_img` which would then replace it in `_set_spatial_attrs` although I'm against adding yet another parameter. Would be curious to hear your thoughts.